cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -493,6 +508,15 @@ public ProgramController start(ProgramId programId, Map<String, String> override
       userArgs.putAll(overrides);
     }
 
+    if ((userArgs.containsKey(RUNTIME_ARG_KEYTAB)) &&
```

Comment:
This logic is repeated. Please refactor.

Reply:
Fixed